### PR TITLE
Added author interaction.

### PR DESCRIPTION
**Pull request by urvishp80**
1. Added option to choose author when getting an answer, this will use resources from that author to form the answer.
2. Updated UI with a checkbox view to select author. UI will be hidden when the mobile view is used.

### CODE REVIEW

**Review by adamjonas — COMMENTED** [review]
You should only be allowed to speak with one author at a time. I understand checkboxes could easily be changed to a radio button but the concept of multiple authors is being processed as an AND rather than an OR. I'm not sure which is right but we should lock it down to one author.

Also, we discussed adding to the prompt the instruction to answer in the style of the author. This may require filtering out transcriptions since they don't reflect the writing style which is a shame because there is probably some good info in there. Maybe something can be done to extract the style from written sources and information from everything?

Another thing I'm seeing is less than the 6 results. What is the method this uses to query results based on the author?

For example, Greg Maxwell had a big hand in inventing taproot. [This doesn't have much](https://bitcoinsearch.xyz/?q=taproot&size=n_50_n&filters%5B0%5D%5Bfield%5D=authors&filters%5B0%5D%5Bvalues%5D%5B0%5D=Gregory%20Maxwell&filters%5B0%5D%5Btype%5D=all), but it has some. Why would chatbtc not be able to find those?

Unrelated to this, gmaxwell and greg maxwell are the same person, but I assume they aren't being linked as [described here](https://github.com/bitcoinsearch/synonyms#readme). Does this need to be setup on the index or is the query not taking advantage of the analyzer?

File: src/pages/api/chat.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ export default async function handler(req: Request): Promise<Response> {
     }
 
     // const keywords = await extractKeywords(question);
-
```
[the remark this answers is not in the document]
Comment (urvishp80) [reply src/pages/api/chat.ts:33]: ok

**Comment by urvishp80** [comment]
> You should only be allowed to speak with one author at a time. I understand checkboxes could easily be changed to a radio button but the concept of multiple authors is being processed as an AND rather than an OR. I'm not sure which is right but we should lock it down to one author.
> 
> Also, we discussed adding to the prompt the instruction to answer in the style of the author. This may require filtering out transcriptions since they don't reflect the writing style which is a shame because there is probably some good info in there. Maybe something can be done to extract the style from written sources and information from everything.
> 
> Another thing I'm seeing is less than the 6 results. What is the method this uses to query results based on the author?
> 
> For example, Greg Maxwell had a big hand in inventing taproot. [This doesn't have much](https://bitcoinsearch.xyz/?q=taproot&size=n_50_n&filters%5B0%5D%5Bfield%5D=authors&filters%5B0%5D%5Bvalues%5D%5B0%5D=Gregory%20Maxwell&filters%5B0%5D%5Btype%5D=all), but it has some. Why would chatbtc not be able to find those?
> 
> Unrelated to this, gmaxwell and greg maxwell are the same person, but I assume they aren't being linked as [described here](https://github.com/bitcoinsearch/synonyms#readme). Does this need to be set up on the index or is the query not taking advantage of the analyzer?

yeah understood. I think I can add the analyzer but we are not using that. I can update the UI as well with single author selection, but I thought like bitcoinsearch we are going to have more options but yeah I understand.

**Comment by urvishp80** [comment]
Closing the PR. Will open another one from fresh branch

**Pull request closed without merging**